pkg/autoscaler/statforwarder/processor.go

```diff
@@ -54,11 +74,49 @@ func (p *bucketProcessor) process(sm asmetrics.StatMessage) error {
 		return err
 	}
 
-	return p.conn.SendRaw(gorillawebsocket.BinaryMessage, b)
+	if p.podAddressable && p.podConn.SendRaw(gorillawebsocket.BinaryMessage, b) == nil {
+		// Pod is accessible via IP address, close the connection via SVC.
+		if p.svcConn != nil {
+			if err := p.svcConn.Shutdown(); err != nil {
+				p.logger.Warnw("Failed to close connection", zap.Error(err))
+			} else {
+				p.svcConn = nil
```

Comment:
I think you want this anyway. I don't think the actual connection is usable after we called `Shutdown` on it.